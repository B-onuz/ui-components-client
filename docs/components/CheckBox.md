# CheckBox (Component)

Componente de formulário usado para valores booleanos.

## Como utilizar

```js
import CheckBox from 'ui-components/lib/components/CheckBox'
```

```jsx
<CheckBox id="tributario" label="Aceite os termos e condições" />
```

## Variações

### Checkbox não controlado

```jsx
import React from 'react'
import CheckBox from 'ui-components/lib/components/CheckBox'

const UncontrolledCheckbox = () => {
  return <CheckBox id="tributario" label={'Aceite os termos e condições'} />
}
```

### Checkbox controlado

```jsx
import React, { useState } from 'react'
import CheckBox from 'ui-components/lib/components/CheckBox'

const ControlledCheckbox = () => {
  const [acceptConditions, setAcceptConditions] = useState(false)

  const handleChangeAcceptConditions = (event) => {
    const target = event.target
    setAcceptConditions(target.checked)
  }

  return <CheckBox id="tributario" checked={acceptConditions} onChange={handleChangeAcceptConditions} label={'Aceite os termos e condições'} />
}
```
