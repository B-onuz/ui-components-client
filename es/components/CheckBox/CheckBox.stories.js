'use strict'

function _typeof(obj) {
  '@babel/helpers - typeof'
  if (typeof Symbol === 'function' && typeof Symbol.iterator === 'symbol') {
    _typeof = function _typeof(obj) {
      return typeof obj
    }
  } else {
    _typeof = function _typeof(obj) {
      return obj && typeof Symbol === 'function' && obj.constructor === Symbol && obj !== Symbol.prototype ? 'symbol' : typeof obj
    }
  }
  return _typeof(obj)
}

Object.defineProperty(exports, '__esModule', {
  value: true,
})
exports.ControlledCheckbox = exports.CheckBoxExample = exports['default'] = void 0

var _react = _interopRequireWildcard(require('react'))

var _CheckBox = _interopRequireDefault(require('./CheckBox'))

var _styledComponents = require('reflexbox/styled-components')

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/CheckBox/CheckBox.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

function _getRequireWildcardCache() {
  if (typeof WeakMap !== 'function') return null
  var cache = new WeakMap()
  _getRequireWildcardCache = function _getRequireWildcardCache() {
    return cache
  }
  return cache
}

function _interopRequireWildcard(obj) {
  if (obj && obj.__esModule) {
    return obj
  }
  if (obj === null || (_typeof(obj) !== 'object' && typeof obj !== 'function')) {
    return { default: obj }
  }
  var cache = _getRequireWildcardCache()
  if (cache && cache.has(obj)) {
    return cache.get(obj)
  }
  var newObj = {}
  var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor
  for (var key in obj) {
    if (Object.prototype.hasOwnProperty.call(obj, key)) {
      var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null
      if (desc && (desc.get || desc.set)) {
        Object.defineProperty(newObj, key, desc)
      } else {
        newObj[key] = obj[key]
      }
    }
  }
  newObj['default'] = obj
  if (cache) {
    cache.set(obj, newObj)
  }
  return newObj
}

var _default = {
  title: 'CheckBox',
  component: _CheckBox['default'],
}
exports['default'] = _default

var CheckBoxExample = function CheckBoxExample() {
  return /*#__PURE__*/ _react['default'].createElement(
    _styledComponents.Flex,
    {
      flexWrap: 'wrap',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 11,
        columnNumber: 3,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(
      _styledComponents.Box,
      {
        width: [1, 1 / 2, 1 / 3],
        m: 2,
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 12,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(_CheckBox['default'], {
        id: 'tributarioa',
        label: 'Tributário',
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 13,
          columnNumber: 7,
        },
      })
    ),
    /*#__PURE__*/ _react['default'].createElement(
      _styledComponents.Box,
      {
        width: [1, 1 / 2, 1 / 3],
        m: 2,
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 15,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(_CheckBox['default'], {
        id: 'tributariob',
        label: 'Tributário',
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 16,
          columnNumber: 7,
        },
      })
    ),
    /*#__PURE__*/ _react['default'].createElement(
      _styledComponents.Box,
      {
        width: [1, 1 / 2, 1 / 3],
        m: 2,
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 18,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(_CheckBox['default'], {
        id: 'tributarioc',
        label: 'Tributário',
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 19,
          columnNumber: 7,
        },
      })
    )
  )
}

exports.CheckBoxExample = CheckBoxExample

var ControlledCheckbox = function ControlledCheckbox() {
  var _useState = (0, _react.useState)(false),
    acceptConditions = _useState[0],
    setAcceptConditions = _useState[1]

  var handleChangeAcceptConditions = function handleChangeAcceptConditions(event) {
    var target = event.target
    setAcceptConditions(target.checked)
  }

  return /*#__PURE__*/ _react['default'].createElement(_CheckBox['default'], {
    id: 'tributarioa',
    checked: acceptConditions,
    onChange: handleChangeAcceptConditions,
    label: 'Aceite os termos e condições',
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 32,
      columnNumber: 10,
    },
  })
}

exports.ControlledCheckbox = ControlledCheckbox
