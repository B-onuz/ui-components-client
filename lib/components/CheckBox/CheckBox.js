"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/CheckBox/CheckBox.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n  &:checked {\n    & ~ ", " {\n      border: 1px solid #95c8da;\n      ", " {\n        animation: start 0.4s ease both;\n        animation-play-state: running !important;\n        width: 10px;\n        height: 5px;\n        border: solid 2px #0c0c0c;\n        border-top: none;\n        border-right: none;\n      }\n    }\n  }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  margin: auto;\n  left: -2px;\n  right: 0;\n  top: calc(50% + 40);\n  width: 0;\n  height: 0;\n  border: none;\n  transform: rotate(-45deg);\n  transform-origin: top left;\n  margin-top: 10px;\n  @keyframes start {\n    0% {\n      opacity: 0;\n      width: 0;\n      height: 0;\n    }\n\n    50% {\n      opacity: 1;\n      width: 0;\n      height: 5px;\n    }\n    100% {\n      width: 10px;\n    }\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  cursor: pointer;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  display: inline-block;\n  width: 17px;\n  height: 17px;\n  border: 1px solid #d4d4d5;\n  border-radius: 4px;\n  margin-right: 10px;\n  position: relative;\n  flex-shrink: 0;\n  cursor: pointer;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  display: flex;\n  align-items: flex-start;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledCheckBox = _styledComponents["default"].div(_templateObject());

var FakeCheckBox = _styledComponents["default"].label(_templateObject2());

var CheckboxLabel = _styledComponents["default"].label(_templateObject3());

var CheckIcon = _styledComponents["default"].div(_templateObject4());

var RealCheckBox = _styledComponents["default"].input(_templateObject5(), FakeCheckBox, CheckIcon);

var CheckBox = function CheckBox(_ref) {
  var children = _ref.children,
      id = _ref.id,
      label = _ref.label,
      rest = _objectWithoutPropertiesLoose(_ref, ["children", "id", "label"]);

  return /*#__PURE__*/_react["default"].createElement(StyledCheckBox, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 74,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react["default"].createElement(RealCheckBox, _extends({
    id: id,
    type: "checkbox",
    style: {
      display: 'none'
    }
  }, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 75,
      columnNumber: 7
    }
  })), /*#__PURE__*/_react["default"].createElement(FakeCheckBox, {
    htmlFor: id,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 76,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react["default"].createElement(CheckIcon, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 77,
      columnNumber: 9
    }
  })), /*#__PURE__*/_react["default"].createElement(CheckboxLabel, {
    htmlFor: id,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 79,
      columnNumber: 7
    }
  }, label));
};

var _default = CheckBox;
exports["default"] = _default;
module.exports = exports.default;