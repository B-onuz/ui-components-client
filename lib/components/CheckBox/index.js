"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _CheckBox = _interopRequireDefault(require("./CheckBox"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _CheckBox["default"];
exports["default"] = _default;
module.exports = exports.default;