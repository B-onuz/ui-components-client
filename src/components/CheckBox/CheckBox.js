import React from 'react'
import styled from 'styled-components'

const StyledCheckBox = styled.div`
  display: flex;
  align-items: flex-start;
`

const FakeCheckBox = styled.label`
  display: inline-block;
  width: 17px;
  height: 17px;
  border: 1px solid #d4d4d5;
  border-radius: 4px;
  margin-right: 10px;
  position: relative;
  flex-shrink: 0;
  cursor: pointer;
`

const CheckboxLabel = styled.label`
  cursor: pointer;
`

const CheckIcon = styled.div`
  position: absolute;
  margin: auto;
  left: -2px;
  right: 0;
  top: calc(50% + 40);
  width: 0;
  height: 0;
  border: none;
  transform: rotate(-45deg);
  transform-origin: top left;
  margin-top: 10px;
  @keyframes start {
    0% {
      opacity: 0;
      width: 0;
      height: 0;
    }

    50% {
      opacity: 1;
      width: 0;
      height: 5px;
    }
    100% {
      width: 10px;
    }
  }
`

const RealCheckBox = styled.input`
  &:checked {
    & ~ ${FakeCheckBox} {
      border: 1px solid #95c8da;
      ${CheckIcon} {
        animation: start 0.4s ease both;
        animation-play-state: running !important;
        width: 10px;
        height: 5px;
        border: solid 2px #0c0c0c;
        border-top: none;
        border-right: none;
      }
    }
  }
`

const CheckBox = ({ children, id, label, ...rest }) => {
  return (
    <StyledCheckBox>
      <RealCheckBox id={id} type="checkbox" style={{ display: 'none' }} {...rest} />
      <FakeCheckBox htmlFor={id}>
        <CheckIcon />
      </FakeCheckBox>
      <CheckboxLabel htmlFor={id}>{label}</CheckboxLabel>
    </StyledCheckBox>
  )
}

export default CheckBox
