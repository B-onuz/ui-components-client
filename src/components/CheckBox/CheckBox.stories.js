import React, { useState } from 'react'
import CheckBox from './CheckBox'
import { Flex, Box } from 'reflexbox/styled-components'

export default {
  title: 'CheckBox',
  component: CheckBox,
}

export const CheckBoxExample = () => (
  <Flex flexWrap={'wrap'}>
    <Box width={[1, 1 / 2, 1 / 3]} m={2}>
      <CheckBox id="tributarioa" label={'Tributário'} />
    </Box>
    <Box width={[1, 1 / 2, 1 / 3]} m={2}>
      <CheckBox id="tributariob" label={'Tributário'} />
    </Box>
    <Box width={[1, 1 / 2, 1 / 3]} m={2}>
      <CheckBox id="tributarioc" label={'Tributário'} />
    </Box>
  </Flex>
)

export const ControlledCheckbox = () => {
  const [acceptConditions, setAcceptConditions] = useState(false)

  const handleChangeAcceptConditions = (event) => {
    const target = event.target
    setAcceptConditions(target.checked)
  }

  return <CheckBox id="tributarioa" checked={acceptConditions} onChange={handleChangeAcceptConditions} label={'Aceite os termos e condições'} />
}
