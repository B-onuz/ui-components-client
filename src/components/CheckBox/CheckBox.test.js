import React from 'react'
import CheckBox from './CheckBox.js'
import renderer from 'react-test-renderer'

test('CheckBox changes the class when hovered', () => {
  const component = renderer.create(<CheckBox />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
