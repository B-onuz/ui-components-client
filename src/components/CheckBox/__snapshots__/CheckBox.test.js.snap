// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox changes the class when hovered 1`] = `
<div
  className="sc-AxjAm ldmlCS"
>
  <input
    className="sc-AxhUy csnest"
    style={
      Object {
        "display": "none",
      }
    }
    type="checkbox"
  />
  <label
    className="sc-AxirZ iDmfPL"
  >
    <div
      className="sc-AxhCb kmoLeD"
    />
  </label>
  <label
    className="sc-AxiKw gMUReu"
  />
</div>
`;
